src/lib/pages/pools/components/unsupportedSection/index.tsx

```diff
@@ -158,13 +161,27 @@ export const UnsupportedSection = ({
             )}
           </Flex>
           <Flex gap="2" alignItems="center">
-            <Button variant="outline-gray" size="sm">
-              Expand All
+            <Button
+              variant="outline-gray"
+              w="94px"
+              size="sm"
+              onClick={() =>
+                setExpandedIndex((prev) =>
+                  prev.length ? [] : Array.from(Array(pageSize).keys())
+                )
+              }
+            >
+              {expandedIndex.length ? "Collapse All" : "Expand All"}
             </Button>
           </Flex>
         </Flex>
       </Flex>
-      <UnsupportedPoolList pools={pools} isLoading={isLoading} />
+      <UnsupportedPoolList
```

Comment:
To reduce the complexity of type involved, I prefer to pass a function that wraps `setExpandedIndex`, like this.
```tsx
  const updateExpandedIndex = (indexes: number[]) => {
    setExpandedIndex(indexes);
  };
```
and pass as props on `UnsupportedPoolList`
```tsx
setExpandedIndex={updateExpandedIndex}
```